Release post for 4.17.0 (#6601)
fleetdm/fleet: articles/fleet-4.17.0.md

# Fleet 4.17.0 | Better osquery management, user engagement, improved host vitals.

![Fleet 4.17.0](../website/assets/images/articles/4.17.0-cover-1600x900@2x.jpg)

Fleet 4.17.0 is up and running. Check out the full [changelog](https://github.com/fleetdm/fleet/releases/tag/fleet-v4.17.0) or continue reading to get the highlights.

For update instructions, see our [upgrade guide](https://fleetdm.com/docs/deploying/upgrading-fleet) in the Fleet docs.

## Highlights
- Manage osquery across your organization. 
- Engage with your users through Fleet Desktop.

> - Battery condition added to **Host Details** for MacOS.

Reviewer: ```suggestion
- Battery condition added to **Host Details** for macOS.
```

Very small nit.

## Manage osquery across your organization.
**Available in Fleet Free and Fleet Premium**

![Manage osquery across your organization..](../website/assets/images/articles/fleet-4.17.0-1-1600x900@2x.jpg)

> Fleet's osquery installer is out of beta! Orbit is a lightweight wrapper for osquery that automates the process of installing, updating and configuring osquery on all of your devices. Even if you're not using Fleet!

Reviewer: ```suggestion
Fleet's osquery installer (Orbit) is out of beta! Orbit is a lightweight wrapper for osquery that automates the process of installing, updating and configuring osquery on all of your devices. Even if you're not using Fleet!
```

Maybe add Orbit in parenthesis here? This way, the user isn't confused by seeing "Orbit" in the next sentence. This makes sense if Fleet's osquery installer is equivalent to Orbit. I'm not totally sure..

> Future plans for the installer include adding the ability to manage `osquery` extensions and enrollment flags across your organization without needing to reinstall anything or ssh in to individual machines. A simple, one-stop shop for remotely managing `osquery`.

Reviewer: Love this paragraph.

> You can generate your installer using [`fleetctl package`](https://fleetdm.com/docs/deploying/fleetctl-agent-updates#packaging-with-orbit). If you're enrolling your host to a Fleet server, he best place to find the full command with flags specific to your Fleet instance is **Hosts > Add hosts** in the Fleet UI. Once the installer has been generated, just extract the file and run it on the host!

Reviewer: ```suggestion
You can generate your installer using the [`fleetctl`] CLI tool. (https://fleetdm.com/docs/deploying/fleetctl-agent-updates#packaging-with-orbit). If you're enrolling your host to a Fleet server, he best place to find the full command with flags specific to your Fleet instance is **Hosts > Add hosts** in the Fleet UI. Once the installer has been generated, just extract the file and run it on the host!
```

Suggested change to specify that fleetctl is a command line tool. This is because the user users fleetctl to generate the package. They run the `fleetctl package` command. I really could be overthinking this. Up to you

> You can generate your installer using the [`fleetctl`] CLI tool. (https://fleetdm.com/docs/deploying/fleetctl-agent-updates#packaging-with-orbit). If you're enrolling your host to a Fleet server, he best place to find the full command with flags specific to your Fleet instance is **Hosts > Add hosts** in the Fleet UI. Once the installer has been generated, just extract the file and run it on the host!

Reviewer: ```suggestion
You can generate your installer using the [`fleetctl` CLI tool](https://fleetdm.com/docs/using-fleet/adding-hosts#osquery-installer). If you're enrolling your host to a Fleet server, he best place to find the full command with flags specific to your Fleet instance is **Hosts > Add hosts** in the Fleet UI. Once the installer has been generated, just extract the file and run it on the host!
```

Reviewer: @ksatter it looks like the "Osquery installer" section is more up to date. 

What do you think about also updating the "Deploying > osquery installer" section to link to the "Adding hosts > Osquery installer" section? This way we can remove some duplication.

Author: Sounds good! GitHub seems to be glitching a bit and seems to have made the commit without resolving, so I'm manually resolving this.

Author: It looks like we've got very similar documentation in three places and it makes sense in each place. I'm adding this to DevRel office hours for discussion.

Reviewer: > I'm adding this to DevRel office hours for discussion.

Great! Ty.

## Battery condition added to Host Details for MacOS.
**Available in Fleet Free & Fleet Premium**

![Battery condition added to Host Details.](../website/assets/images/articles/fleet-4.17.0-3-1600x900@2x.jpg)
Quickly determine whether a MacOS host's battery needs to be replaced. The physical health of your hosts is every bit as important as security and Fleet makes it easy to see how well the battery is performing. 

## More new features, improvements, and bug fixes

In 4.17.0, we also:

* Added native support for M1 Macs.

* Improved reporting of error states on the health dashboard and added separate health checks for MySQL and Redis with `/healthz?check=mysql` and `/healthz?check=redis`.

* Improved SSO login failure messaging.

* Fixed osquery tables that report incorrect platforms.

* Added `docker_container_envs` table to the osquery table schema on the **Query* page.

* Updated Fleet host detail query so that the `os_version` for Ubuntu hosts reflects the accurate patch number.

* Improved accuracy of `software_host_counts` by removing hosts from the count if software has been uninstalled.

* Improved accuracy of the `last_restarted` date. 

> * Fixed `/api/_version_/fleet/hosts/identifier/{identifier}` to return the correct value for `host.status`.

Reviewer: ```suggestion
* Fixed `/api/v1/fleet/hosts/identifier/{identifier}` to return the correct value for `host.status`.
```

Suggestion to use `v1` here. If this makes sense, can you please also update the CHANGELOG entry for this?

Reasoning on waiting to include (publicly announce) the new API versioning is here: https://github.com/fleetdm/fleet/pull/6137#issuecomment-1175530387
